src/main/java/pl/edu/agh/mwo/reporter/report/printer/ExcelExporter.java

```diff
@@ -1,39 +1,67 @@
 package pl.edu.agh.mwo.reporter.report.printer;
 
+
 import org.apache.poi.EncryptedDocumentException;
+import org.apache.poi.hssf.usermodel.HSSFSheet;
+import org.apache.poi.hssf.usermodel.HSSFWorkbook;
+import org.apache.poi.openxml4j.exceptions.InvalidFormatException;
 import org.apache.poi.ss.usermodel.Cell;
 import org.apache.poi.ss.usermodel.Row;
-import org.apache.poi.xssf.usermodel.XSSFSheet;
-import org.apache.poi.xssf.usermodel.XSSFWorkbook;
 
+import java.io.File;
+import java.io.FileInputStream;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.math.BigDecimal;
 
 public class ExcelExporter {
 
-    private final XSSFWorkbook workbook;
-
-    private final XSSFSheet sheet;
+    private HSSFWorkbook workbook;
 
-    private final String excelFilePath;
+    private HSSFSheet sheet;
 
     private int rowsCount;
 
     private Row lastRow;
 
-    public ExcelExporter(String excelFilePath, String reportName, String title, String[] headers) {
-        this.excelFilePath = excelFilePath;
-        workbook = new XSSFWorkbook();
-        sheet = workbook.createSheet(reportName);
-
-        addRow();
-        addCell(0, title);
+    public boolean openFile(String excelFilePath) {
+        try {
+            File fileOut = new File(excelFilePath);
+
+            if (fileOut.exists()){
+                FileInputStream fileInputStream = new FileInputStream(excelFilePath);
+                workbook = new HSSFWorkbook(new FileInputStream(fileOut));
+                sheet = workbook.getSheetAt(0);
+            //    workbook.close();
+
+            } else {
+                workbook = new HSSFWorkbook();
+                sheet = workbook.createSheet("Report1");
+
+            }
+        } catch (EncryptedDocumentException | IOException e ) {
+            System.out.println("Błąd przy otwieraniu pliku: " + excelFilePath);
+            return false;
+        }
+//        catch (InvalidFormatException e){
+//            System.out.println("Niewłaściwy format pliku: " + excelFilePath);
+//        }
+        return true;
+    }
 
-        addRow();
-        addRow();
-        for (int i = 0; i < headers.length; i++) {
-            addCell(i, headers[i]);
+    public ExcelExporter(String excelFilePath, String reportName, String title, String[] headers) {
+        boolean flag;
+        System.out.println();
+        flag = openFile(excelFilePath);
+        if (flag) {
```

Comment:
Może lepiej zadeklarowac, że konstruktor rzuca wyjątek i obsłużyć to poziom wyżej.